src/bp/core/services/dialog/decision-engine.ts

```diff
@@ -55,23 +55,24 @@ export class DecisionEngine {
         this.onAfterEventProcessed && (await this.onAfterEventProcessed(event))
       }
     }
-
-    for (const { action, data } of event.ndu.actions) {
-      if (action === 'send' && data) {
-        const content = data as NDU.SendContent
-        await this._sendContent(content, event)
-
-        BOTPRESS_CORE_EVENT('bp_core_send_content', {
-          botId: event.botId,
-          channel: event.channel,
-          source: content.source,
-          details: content.sourceDetails!
-        })
-      } else if (action === 'redirect' || action === 'startWorkflow' || action === 'goToNode') {
-        const { flow, node } = data as NDU.FlowRedirect
-        const flowName = flow.endsWith('.flow.json') ? flow : `${flow}.flow.json`
-
-        await this.dialogEngine.jumpTo(sessionId, event, flowName, node)
+    if (!event.restored) {
```

Comment:
why not use a guard statement like everywhere else ? 

```
if(event.restored) {
  return
}
```

Reply:
Because I only want to avoid processing the NDU actions there, but keep the rest of the processing. (so we continue where we were previously) Otherwise NDU will cause the prompts to be processed again